core/src/main/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java

```diff
@@ -92,7 +91,7 @@ public boolean followedOrderByAdvice() {
     public PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext, int order) throws SqlException {
         DataFrameCursor dataFrameCursor = dataFrameCursorFactory.getCursor(executionContext, order);
         if (framingSupported) {
-            if (order == ORDER_ASC || order == ORDER_ANY) {
+            if (order == ORDER_ASC || (order == ORDER_ANY && dataFrameCursorFactory.getOrder() != ORDER_DESC)) {
```

Comment:
I'd better keep this method as is and apply the fix to `AsyncFilteredRecordCursorFactory/AsyncJitFilteredRecordCursorFactory#getCursor()`. It should consider `base.hasDescendingOrder()` and also split the direction record traversal (ORDER BY) and limit accumulation (LIMIT). Currently they're mixed which leads to bugs and confusion.

Reply:
Fixed.